rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,248 @@
+		},
+	});
+}
+
+class SemverNodeVersion {
+	constructor(nodeVersion) {
+		this.nodeVersion = nodeVersion;
+		this.validNodeVersion = semver.coerce(nodeVersion);
+	}
+
+	compare(featureVersion) {
+		const supportedNodeVersion = semver.coerce(featureVersion);
+		return this.validNodeVersion
+			? semver.lte(supportedNodeVersion, this.validNodeVersion)
+			: semver.ltr(supportedNodeVersion, this.nodeVersion);
+	}
+}
+
+function processRule(context, node, moduleName, targets) {
+	if (!moduleName || typeof moduleName !== 'string' || ['.', '/'].includes(moduleName[0])) {
+		return;
+	}
+
+	const nodeVersion = targets instanceof SemverNodeVersion && targets;
+	const importedModule = moduleName.replace(/([/\\].+?)\.[^.]+$/, '$1');
+
+	const unavailableFeatures = coreJsCompat({targets}).list;
+	const coreJsModuleFeatures = coreJsEntries[importedModule.replace('core-js-pure', 'core-js')];
+
+	const checkFeatures = features => nodeVersion
+		? features.every(feature => compatData[feature].node && nodeVersion.compare(compatData[feature].node))
+		: !features.every(feature => unavailableFeatures.includes(feature));
+
+	if (coreJsModuleFeatures) {
+		if (coreJsModuleFeatures.length > 1) {
+			if (checkFeatures(coreJsModuleFeatures)) {
+				context.report({
+					node,
+					messageId: MESSAGE_ID_CORE_JS,
+					data: {
+						coreJsModule: moduleName,
+					},
+				});
+			}
+		} else if (nodeVersion ? nodeVersion.compare(compatData[coreJsModuleFeatures[0]].node) : !unavailableFeatures.includes(coreJsModuleFeatures[0])) {
+			report(context, node, coreJsModuleFeatures[0]);
+		}
+
+		return;
+	}
+
+	const polyfill = polyfills.find(({pattern}) => pattern.test(importedModule));
+	if (polyfill) {
+		const [, namespace, method = ''] = polyfill.feature.split('.');
+		const [, features] = Object.entries(coreJsEntries).find(it => it[0] === `core-js/actual/${namespace}${method && '/'}${method}`);
+		if (checkFeatures(features)) {
+			report(context, node, polyfill.feature);
+		}
+	}
+}
+
+function create(context) {
+	const options = context.options[0];
+	let targets = options && options.targets;
+	if (!targets) {
+		targets = getTargetsFromPackage(context.getFilename(), options && options.useBrowserlistFieldByDefault);
+	}
+
+	if (!targets) {
+		return {};
+	}
+
+	return {
+		[STATIC_REQUIRE_SOURCE_SELECTOR](node) {
+			processRule(context, node, node.value, targets);
+		},
+		':matches(ImportDeclaration, ImportExpression)[source.type="Literal"]'(node) {
+			processRule(context, node, node.source.value, targets);
+		},
+	};
+}
+
+const schema = [
+	{
+		type: 'object',
+		additionalProperties: false,
+		required: ['targets'],
+		properties: {
+			useBrowserlistFieldByDefault: {type: 'boolean'},
+			targets: {
+				oneOf: [
+					{
+						type: 'string',
+						minLength: 1,
+					},
+					{
+						type: 'array',
+						minItems: 1,
+						items: {
+							type: 'string',
+						},
+					},
+					{
+						type: 'object',
+						minProperties: 1,
+						properties: {
```

Comment:
Let's not list the properties, we don't want to maintain this, or even not list "string"/"array", just pass the `targets` through.